vlad.metelap/task1.rb

```diff
@@ -0,0 +1,30 @@
+class Task1
+
+  b = Array.new(3, Array.new(3, 0))
+  # b = [[0,0,0],[0,0,0],[0,0,0]]
+  b = b.map {|el| el.map {rand(-10..10)}}
+  positivearr = b.map {|el| el.map {|elm| elm > 0 ? elm : 0} }
+
+  p "Task 1"
+  p "Array 01: " + b.inspect
+  b.select {|el| el.select {|elm| puts "#{elm}" if elm > 0} }
+  amount = 0
+  positivearr.select {|el| el.select {|elm| amount += elm } }
+
+  p "Array 02: " + positivearr.inspect
+  p "Amount: #{amount}"
+  p "***********************************************************"
+  p "Task 2"
+  h = {'Yes' => 23, 'b' => 'travel', 'yesterday' => 34, 5 => '234', yesss: :fg, try: 30, key: 'some value',
+       ' yesterday1 '=> 34,' yesteryear '=> 2014}
+  p "Hash 01: " + h.inspect
+  number_elements = 0
```

Comment:
It would be better to avoid having an extra iterator variable `number_elements `.
As alternative I recommend you to realize you task with methods `filter_map` if you use ruby ` 2.7.0+` (or `map` + `compact` otherwise) and `count`


https://ruby-doc.org/core-2.7.0/Enumerable.html#method-i-filter_map
https://apidock.com/ruby/Array/count
https://apidock.com/ruby/Array/map
https://apidock.com/ruby/v2_5_5/Array/compact

Reply:
done